src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -197,6 +197,7 @@
                   (.setKeyStore (:keystore keystore-config))
                   (.setKeyStorePassword (:key-password keystore-config))
                   (.setTrustStore (:truststore keystore-config))
+                  (.setEndpointIdentificationAlgorithm nil)
```

Comment:
So, Jetty's preferred way to handle this issue is actually to create an instance of `SslContextFactory.Server` (which is a static inner class that will have the correct configuration).

However we are currently using `InternalSslContextFactory` which is our _own_ subclass, that handles how we reload the CRL.

In Jetty 10, SslContextFactory is becoming an abstract class and its methods are currently deprecated. I'm unsure if now is the correct time to rework that subclassing, and if so, the best way to go forward.

Reply:
I looked into this a little bit more and didn't see an obvious way forward besides what I'm doing here (but I'm not a Java expert and would love someone to double check me on this).

The concern about changes to Jetty 10 are probably moot considering this is "trapperkeeper-webserver-jetty _9_ ".

Reply:
Suggest maybe adding a bit of explanation about the rationale/effect of this change to the commit message.

Reply:
`setEndpointIdentificationAlgorithm` is a client-side thing to validate that the certificate matches the host that sent it (X509v3 extensions, etc) - it should be safe to make this nil in a server.

As far as reloading the CRL, I guess we get to rework our class when Jetty 10 happens.